tests/test_all_rules.py

```diff
@@ -513,6 +515,105 @@ def test_integration_tag(self):
                 """
             self.fail(err_msg + '\n'.join(failures))
 
+    def test_invalid_queries(self):
+        # invalid queries
+        invalid_queries = [
+            """
+            process where process.fake == "cmd.exe"
+            """
+        ]
+        invalid_integration_queries = [
+            """file where event.dataset == "google_workspace.fake" and event.action : ("copy", "view", "download") and
+                    google_workspace.fake: "people_with_link" and source.user.email == "" and
+                    file.fake: (
+                        "token","assig", "pssc", "keystore", "pub", "pgp.asc", "ps1xml", "pem", "gpg.sig", "der", "key",
+                        "p7r", "p12", "asc", "jks", "p7b", "signature", "gpg", "pgp.sig", "sst", "pgp", "gpgz", "pfx",
+                        "p8", "sig", "pkcs7", "jceks", "pkcs8", "psc1", "p7c", "csr", "cer", "spc", "ps2xml")
+            """
+        ]
+
+        valid_queries = [
+            """file where file.extension: (
+                "token","assig", "pssc", "keystore", "pub", "pgp.asc", "ps1xml", "pem",
+                "p7r", "p12", "asc", "jks", "p7b", "signature", "gpg", "pgp.sig", "sst",
+                "p8", "sig", "pkcs7", "jceks", "pkcs8", "psc1", "p7c", "csr", "cer")
+            """,
+            """file where event.dataset == "google_workspace.drive" and event.action : ("copy", "view", "download") and
+                    google_workspace.drive.visibility: "people_with_link" and source.user.email == "" and
+                    file.extension: (
+                        "token","assig", "pssc", "keystore", "pub", "pgp.asc", "ps1xml", "pem", "gpg.sig", "der", "key",
+                        "p7r", "p12", "asc", "jks", "p7b", "signature", "gpg", "pgp.sig", "sst", "pgp", "gpgz", "pfx",
+                        "p8", "sig", "pkcs7", "jceks", "pkcs8", "psc1", "p7c", "csr", "cer", "spc", "ps2xml")
+            """
+
+        ]
+
+        base_fields = {
+            "author": ["Elastic"],
+            "description": "test description",
+            "index": ["filebeat-*"],
+            "language": "eql",
+            "license": "Elastic License v2",
+            "name": "test rule",
+            "risk_score": 21,
+            "rule_id": str(uuid.uuid4()),
+            "severity": "low",
+            "type": "eql"
+        }
```

Comment:
is there no clean way to reuse the logic from `test_eql_validation`?

Reply:
I am not sure. Given that is is locally scoped to `test_eql_validation()`, I think we would have to move `base_fields` somewhere that both `test_eql_validation()` and `test_invalid_queries()` could reference it in order to not duplicate code. This could also be done for the `build_rule()` function to further reduce code replication.

Reply:
Leaving this as duplicated for now. This could be addressed by using fixtures from pytests. However, given some of the intricacies of the unit testing that is currently using the `unittest.TestCase` class, the de-duplication is out of the scope of this issue and will be address in a later issue/FR.